api/machines.go

```diff
@@ -64,6 +83,59 @@ func (mr *machinesResource) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 	sendResponse(rw, http.StatusOK, page)
 }
 
+func (mr *machinesResource) patch(rw http.ResponseWriter, req *http.Request) {
+	ops := make([]machineMetadataOp, 0)
+
+	dec := json.NewDecoder(req.Body)
+
+	if err := dec.Decode(&ops); err != nil {
+		sendError(rw, http.StatusInternalServerError, err)
+		return
+	}
+
+	// Validate all operations before applying
+	for _, op := range ops {
+		if op.Op != "add" && op.Op != "remove" && op.Op != "replace" {
+			sendError(rw, http.StatusBadRequest, fmt.Errorf("Bad op: %s. Expected add/remove/replace", op.Op))
```

Comment:
Simplify this error message, too:

`errors.New("invalid op: expect add, remove or replace")`